### DISCUSSION TITLE
3EX_2EM_pulsed mode - both digital channels as output?

### DISCUSSION THREAD

**Discussion by EmmaLescrauwaet** [pyPhotometry/code#19] [Q&A]
Dear, at our lab we are planning to record simultaneously from 2 sites in a mouse brain, Gcamp in the Locus Coeruleus (+ its isobestic) and Grabne in the hippocampus. To do so, we were planning to use the 3EX_2EM_pulsed mode, in which:

- LED 1 is used for Gcamp, connected to an external LED and the FMC4 cube of Doric lenses 
- Digital 2 works as an output to trigger an external LED driver and LED for the isobestic of Gcamp, connected to the FMC4 cube of Doric lenses 
- LED 2 is used for Grabne, connected to the ilFMC4 cube of Doric Lenses (with integrated LED)
- Analog 1 captures both Gcamp + isobestic Gcamp
- Analog 2 captures Grabne

Is this a correct setup in your opinion? 

Also, we would still need digital 1 is an input to show an external trigger of our treatment on the GUI. Does only digital 2 work as an output when using the 3EX_2EM_pulsed mode, or both digital channels? 

Thank you in advance! 

All the best,
Emma Lescrauwaet

![image](https://github.com/user-attachments/assets/4b07a92b-3841-479c-ad55-e5bf208ffa70)

**Comment by ThomasAkam** [comment]
Hi @EmmaLescrauwaet, 

Apologies for the delay getting back to you, I've just got back from holiday.  The above setup looks sensible to me.  You will only get an isosbestic signal from one of your recording sites though with this setup, if you want to get isosbestic from both sites you could use two pyPhotometry boards, one for each minicube.

When the system is configured in 3EX_2EM_pulsed mode, digital 1 still works as an input as normal.

**Reply by seniasantos** [reply]
Hi,

Thank you @ThomasAkam and @EmmaLescrauwaet, this is what I was looking for! I am building a similar setup to @EmmaLescrauwaet 's, and I'm wondering, in regards to the comment about the 2 pyPhotometry boards (one for each minicube), how do you suggest that I'd sync them up? I am using 2 digital inputs to sync with behavioural data, so I was wondering if perhaps splitting that signal (BNC splitter) could be an option? Also, can I plug the 2 different boards to the same pc and acquire the signals from both cubes real-time, or would I have to have another pc to acquire the data for the second cube?

Many thanks in advance!

**Reply by ThomasAkam** [reply]
Hi @seniasantos,

The simplest way to sync signals recorded from two boards would be to split the sync pulses coming from your behavioural system using a BNC T connector and send them to both of the pyPhotometry board's digital inputs.  This would allow you to align both signals to the same set of sync pulses (see the pyControl [synchronisation docs](https://pycontrol.readthedocs.io/en/latest/user-guide/synchronisation/) for more info on how we use sync pulses for sychronisation between systems).  My one concern with this approach is that the acqusition cycles of turning on and off the LEDs would not be synchonrised between the two pyPhotometry systems, and if the fiber tips are close enough that excitation light from one system can affect the signal recieved by the other, you might get signal variation due to the two systems acqusition cycles drifting in phase relative to each other.  This could be avoided by modifying the code that runs on the pyPhotometry board such that the acqusition cycle on one system is triggered by pulses generated by the other system, which would certainly be possible but would be a bit of work to implement.

You can absolutely plug multiple pyPhotometry boards into a single computer, and can control them both from the same GUI instance.

**Reply by seniasantos** [reply]
Thank you, @ThomasAkam ! Very helpful advice, especially regarding the acquisition cycles of the two systems. I'm gonna check with the team and see the course of action.
